fastapi_utils/base_route.py

```diff
@@ -0,0 +1,109 @@
+from typing import Dict, Generic, List, TypeVar
+
+from fastapi import Depends, HTTPException
+from pydantic import BaseModel
+from sqlalchemy.orm import Session
+
+from fastapi_utils.crud_base import Base, CRUDBase
+
+ResponseModelType = TypeVar("ResponseModelType", bound=BaseModel)
+ResponseModelManyType = TypeVar("ResponseModelManyType", bound=BaseModel)
+CreateSchemaType = TypeVar("CreateSchemaType", bound=BaseModel)
+UpdateSchemaType = TypeVar("UpdateSchemaType", bound=BaseModel)
+IDType = TypeVar("IDType")
+
+
+def get_filter_fields(self) -> List[str]:
+    """This would need to get overridden for each BaseRoute where the filter fields are defined.
+
+    Returns:
+        List[str] -- List of fields to filter by
+    """
+    return []
+
+
+class BaseRoute(Generic[ResponseModelType, ResponseModelManyType, CreateSchemaType, UpdateSchemaType, IDType]):
```

Comment:
Do you think it would make sense to change the name here to `CRUDRoute` rather than `BaseRoute`? `BaseRoute` seems a little overly generic to me.